Added necessary fields like Name for vault and title for Proposal in the existing fields, fixed bugs and minor UI changes

bhavik-mangla opened the pull request:
- **vaultName** for FundingVault
- **title** for Proposal
- Incorporated them in the _UI and create/display files_
- Manually Merged the last PR done by _theishu4_ as I was working on older repo
- Added some minor UI changes while creating space and vault
- Added await when loading vault details

adityabhattad2021 commented on app/web-app/src/blockchain/deployments/production/fairFund_deployment.json:
```diff
@@ -1 +1 @@
-{"fairFund":"0x1BADd20Eed4f56DA23A3544c3c373B18024D8Bd6"}
```
This address should not be changed, as it is testnet deployment address, please do not include this file in the commit.

adityabhattad2021 commented:
- Previous migrations have been deleted in this commit. Please ensure they remain unchanged.  
- There are some files included in the commit that have no relevant changes. Only include files with actual modifications.  
- Please provide a proper PR description that clearly explains every change made in the commit.  
- Additionally, include a short video walkthrough of the changes.  
As far as I remember @theishu4  is also working on this issue (#56). It would be great if you two could collaborate to ensure alignment and avoid any potential conflicts.

bhavik-mangla commented:
Prisma Schema:
Added vaultName to the FundingVault model.
Added title to the Proposal model.

Frontend:
Updated various components to use vaultName instead of vault.id.
Modified forms to include new fields for vaultName and title.
Enhanced form placeholders and added autoComplete attributes.
Changed the rendering of CardWrapper and TableWrapper components to be awaited.

API Routes:
Updated API routes to handle the new vaultName and title fields when creating new FundingVault and Proposal entries.

Migrations:
Added new columns vaultName and title in the database schema for FundingVault and Proposal, respectively.

theishu4 commented on app/web-app/prisma/migrations/0_init/migration.sql:
```diff
@@ -1,6 +1,7 @@
 -- CreateTable
 CREATE TABLE "FundingVault" (
     "id" SERIAL NOT NULL,
+    "vaultName" TEXT NOT NULL,
     "description" TEXT NOT NULL,
     "creatorAddress" TEXT NOT NULL,
     "fundingTokenAddress" TEXT NOT NULL,
```
@bhavik-mangla Please do not edit this file manually or use a hard reset during migration, such as running `prisma db push`. Doing so will cause Prisma to delete all previous migrations and trigger a Drift Detected warning during subsequent migrations.

theishu4 commented on app/web-app/src/components/proposal-form.tsx:
```diff
@@ -282,7 +307,7 @@ export default function ProposalForm({
                             render={({ field }) => {
                                 return (
                                     <FormItem className="col-span-2 md:col-span-1">
-                                        <FormLabel className="flex items-center gap-2">
```
It will cause UI issue
